### PR TITLE
Organize the Spring directory

### PR DESCRIPTION
**Pull request by Netyyyy**
# Description

Organize the Spring directory:
1. move xxx-tests-job.yml into spring/pipeline
2. move xxx-matrix.json into spring/pipeline
3. move spring-cloud-azure-supported-spring.json into spring/pipeline

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/scripts/monitor_update_monitor_matrix_json.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Python version 3.4 or higher is required to run this script.
 #
-# This script is used to update sdk\spring\supported-version-matrix.json before compatibility check.
+# This script is used to update sdk\spring\\pipeline\supported-version-matrix.json before compatibility check.
```
Comment (saragluna) [inline sdk/spring/scripts/monitor_update_monitor_matrix_json.py:3]:
```suggestion
# This script is used to update sdk\spring\pipeline\supported-version-matrix.json before compatibility check.
```

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by saragluna — APPROVED** [review]

**Comment by kushagraThapar** [comment]
@saragluna / @Netyyyy  - when we change any spring-data-cosmos related code, please make sure to run the cosmos ci, and spring-data-cosmos CI pipelines to validate the change. 
Looks like this change broke the cosmos CI pipelines that runs spring data cosmos CI. https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3077931&view=logs&jobId=8bef1c98-23e3-5d36-ff88-ed7605b6f394

on this PR - https://github.com/Azure/azure-sdk-for-java/pull/36687#issuecomment-1717728034 

@saragluna and @Netyyyy , please work with @xinlian12 to fix this issue as quickly as possible so we can get the coverage back up, thanks!

**Comment by Netyyyy** [comment]
> https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3077931&view=logs&jobId=8bef1c98-23e3-5d36-ff88-ed7605b6f394

Fix in https://github.com/Azure/azure-sdk-for-java/pull/36758

**Pull request merged**